documentation/modules/ROOT/pages/configuration/applying-transformations-selectively.adoc

```diff
@@ -0,0 +1,143 @@
+
+// Type: concept
+// ModuleID: about-smt-predicates
+// Title: About SMT predicates
+== SMT predicates
+
+{prodname} provides several single message transformations (SMTs) that you can use to modify event records before Kafka Connect saves the records to Kafka topics.
+By default, when you configure one of these SMTs for a {prodname} connector, Kafka Connect applies that transformation to every record that the connector emits.
+
+However, there can be instances in which you want to apply a transformation selectively, so that it modifies only that subset of change event messages that share a common characteristic.
+For example, you might want to run the transformation only on event messages from a specific table or that include a specific header key.
+In environments that run Kafka 2.6 or greater, you can append a predicate statement to a transformation to instruct Kafka Connect to apply the SMT only to event records that match a specified condition.
+When a predicate is configured, Kafka Connect evaluates each incoming message that it processes against the condition specified in the predicate.
+If the predicate condition is true for the event message, then the SMT is applied.
+Otherwise, the message is passed without any modifications.
+
+After you define a predicate, you can reuse it and apply it to other transforms.
+Predicates also include a `negate` option that you can use to invert a predicate so that the predicate condition is applied only to records that do not match the condition that is defined in the predicate statement.
+You can use the `negate` option to pair the predicate with other transforms that are based on negating the condition.
+
+.Predicate elements
+Predicates include the following elements:
+
+* `predicates` prefix
+* Alias (for example, `isOutboxTable`)
+* Type (for example, `org.apache.kafka.connect.transforms.predicates.TopicNameMatches`).
+  Kafka Connect provides a set of default predicate types, which you can supplement by defining your own custom predicates.
+* Condition statement and any additional configuration properties, depending on the type of predicate (for example, a regex naming pattern)
+
+.Default predicate types
+The following predicate types are available by default:
+
+HasHeaderKey:: Specifies a key name in the header in the event message that you want Kafka Connect to evaluate.
+The predicate evaluates to true for any records that include a header key that has the specified name.
+
+RecordIsTombstone:: Matches records in which the key value of the record is `null`, representing a deleted row in the database.
+Tombstones are fields in a database that indicate that a record is not physically deleted, but has a logical deletion flag that is set to `true` when when the record is marked for deletion.
+The predicate evaluates to `true` for any record that has a `null` key value.
+Use this predicate in combination with a filter SMT to remove tombstone records.
```

Comment:
Probably worth a mention of the `tombstone.on.deletes` option in Debezium which can be set to false, so to not emit tombstones in the first place. A bit more efficient than first producing them and then filtering them out right away. Though it could be useful actually to enable compaction in Kafka (as described above) but filtering out tombstones on the sink side, as some connectors cannot handle them correctly.